### ISSUE TITLE
Segment fault with `--with-gssapi`

### ISSUE THREAD

**Issue by beeender** [greenplum-db/gpdb#11459]
### Greenplum version or build

`master`, reproduced with `server-rc-7.0.0-alpha.0+dev.14209.g8da6059-rhel7_x86_64.tar.gz`

### OS version and uname -a

centos7 docker image
Linux localhost 5.10.11-zen2-1-zen #1 ZEN SMP PREEMPT Wed, 27 Jan 2021 23:13:33 +0000 x86_64 x86_64 x86_64 GNU/Linux

### Installation information ( pg_config )

```
[gpadmin@localhost ~]$ pg_config
BINDIR = /usr/local/greenplum-db-devel/bin
DOCDIR = /usr/local/greenplum-db-devel/share/doc/postgresql
HTMLDIR = /usr/local/greenplum-db-devel/share/doc/postgresql
INCLUDEDIR = /usr/local/greenplum-db-devel/include
PKGINCLUDEDIR = /usr/local/greenplum-db-devel/include/postgresql
INCLUDEDIR-SERVER = /usr/local/greenplum-db-devel/include/postgresql/server
LIBDIR = /usr/local/greenplum-db-devel/lib
PKGLIBDIR = /usr/local/greenplum-db-devel/lib/postgresql
LOCALEDIR = /usr/local/greenplum-db-devel/share/locale
MANDIR = /usr/local/greenplum-db-devel/man
SHAREDIR = /usr/local/greenplum-db-devel/share/postgresql
SYSCONFDIR = /usr/local/greenplum-db-devel/etc/postgresql
PGXS = /usr/local/greenplum-db-devel/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--with-quicklz' '--with-gssapi' '--enable-mapreduce' '--enable-orafce' '--enable-ic-proxy' '--enable-orca' '--enable-gpcloud' '--with-libxml' '--with-pgport=5432' '--enable-cassert' '--enable-tap-tests' '--enable-debug-extensions' '--with-perl' '--with-python' 'PYTHON=python3' '--with-includes=/tmp/build/f8c7ee08/gpdb_src/gpAux/ext/rhel7_x86_64/include /tmp/build/f8c7ee08/gpdb_src/gpAux/ext/rhel7_x86_64/include/libxml2' '--with-libraries=/tmp/build/f8c7ee08/gpdb_src/gpAux/ext/rhel7_x86_64/lib' '--with-openssl' '--with-pam' '--with-ldap' '--disable-rpath' 'LDFLAGS=-Wl,--enable-new-dtags -Wl,-rpath,$ORIGIN/../lib' '--prefix=/usr/local/greenplum-db-devel' '--mandir=/usr/local/greenplum-db-devel/man' 'CFLAGS=-m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g'
CC = gcc
CPPFLAGS = -D_GNU_SOURCE -I/usr/include/libxml2 -I/tmp/build/f8c7ee08/gpdb_src/gpAux/ext/rhel7_x86_64/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-but-set-variable -Werror=implicit-fallthrough=3 -Wno-format-truncation -m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g  -Werror=uninitialized -Werror=implicit-function-declaration
CFLAGS_SL = -fPIC
LDFLAGS = -Wl,--enable-new-dtags -Wl,-rpath,$ORIGIN/../lib -L/tmp/build/f8c7ee08/gpdb_src/gpAux/ext/rhel7_x86_64/lib -Wl,--as-needed
LDFLAGS_EX =
LDFLAGS_SL =
LIBS = -lpgcommon -lpgport -lpthread -lxerces-c -lbz2 -lxml2 -lpam -lrt -lssl -lcrypto -lgssapi_krb5 -luv -lquicklz -lz -lreadline -lrt -lcrypt -ldl -lm  -lcurl   -lzstd
VERSION = PostgreSQL 12beta2
```

### Expected behavior

GPDB can be initialized with kerberos

### Actual behavior

Segment fault

### Step to reproduce the behavior

1. Init the kerberos and call kinit with `gpadmin` user
```
kinit -k -t /var/kerberos/krb5kdc/gpss.keytab gpss/localhost@GPDF.CI
```
2. call `gpinitsystem`
The `postgres` process crashed with below call stack:
```
#0  resetPQExpBuffer (str=str@entry=0x14f95a3) at pqexpbuffer.c:161
#1  0x0000000000a58e67 in pg_GSS_error (mprefix=mprefix@entry=0x14 <error: Cannot access memory at address 0x14>, conn=conn@entry=0x14f91e3, maj_stat=851968,
    min_stat=2529639093) at fe-gssapi-common.c:55
#2  0x0000000000a58b9e in secure_open_gssapi (port=port@entry=0x92922d0) at be-secure-gssapi.c:521
#3  0x0000000000b6400e in ProcessStartupPacket (port=port@entry=0x92922d0, secure_done=secure_done@entry=false) at postmaster.c:2322
#4  0x00000000006dde60 in BackendInitialize (port=0x92922d0) at postmaster.c:4822
#5  BackendStartup (port=0x92922d0) at postmaster.c:4605
#6  ServerLoop () at postmaster.c:1964
#7  0x0000000000b6aa97 in PostmasterMain (argc=argc@entry=9, argv=argv@entry=0x9265290) at postmaster.c:1589
#8  0x00000000006f8ad9 in main (argc=9, argv=0x9265290) at main.c:240
```

### The problem
There are two `pg_GSS_error` exist in both frontend and backend code, the signatures are slightly different:

```
// be-gssapi-common.c
void
pg_GSS_error(int severity, const char *errmsg,
			 OM_uint32 maj_stat, OM_uint32 min_stat)
// fe-gssapi-common.c
void
pg_GSS_error(const char *mprefix, PGconn *conn,
			 OM_uint32 maj_stat, OM_uint32 min_stat)
```
In the GPDB7 merge commit b0b39f72b9904bcb80f97b35837ccff1578aa4b8, the backed one was disabled with `#if 0` and the frontend one is added to the `src/backend/libpq/Makefile`. When ever the code path to `pg_GSS_error()` is triggered, a segment fault will happen, since a `const char*` is passed as a `PGconn *`.

**Comment by lij55** [comment]
Hi @wuj21 , we are working on the GPSS support of GP7 now. It is a blocker for us. I believe Kerberos is also a critical feature for all Greenplum regardless of GPSS.
Could you please help to prioritize it? Thanks in advance.

**Comment by beeender** [comment]
Fixed by #11504
